src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -229,6 +252,21 @@ def safe_to_destroy(self, ids):
                 'is_safe_to_destroy': False,
             }
 
+    @Endpoint('GET', query_params=['svc_ids'])
+    @ReadPermission
+    @raise_if_no_orchestrator
+    @handle_orchestrator_error('osd')
+    def safe_to_delete(self, svc_ids):
+        """
+        :type ids: int|[int]
+        """
+        orch = OrchClient.instance()
+        check = orch.osds.check_remove([svc_ids])
+        return {
+            'is_safe_to_delete': check.get('safe', False),
+            'message': check.get('message', '')
+        }
```

Comment:
I see in the Typescript code that this response is typed as `{is_safe_to_delete: boolean ... }`, instead of `boolean[]`. Does that mean that for multiple OSDs we return a single OK/NOK response? Is that aggregate the result of _&&'ing_ the individual results, or _||'ing_ those?

Reply:
Suppose to logical AND of them, not safe for the whole operation if one of the OSDs is considered not safe. But, right now I only check the status of the cluster (any OSD_NEARFULL or OSD_FULL warning). Individual OSD checks need to be handled in orchestrator.